numpy/core/numerictypes.py

```diff
@@ -339,34 +341,23 @@ def _add_aliases():
 def _add_integer_aliases():
     _ctypes = ['LONG', 'LONGLONG', 'INT', 'SHORT', 'BYTE']
     for ctype in _ctypes:
-        val = typeinfo[ctype]
-        bits = val[2]
-        charname = 'i%d' % (bits//8,)
-        ucharname = 'u%d' % (bits//8,)
-        intname = 'int%d' % bits
-        UIntname = 'UInt%d' % bits
-        Intname = 'Int%d' % bits
-        uval = typeinfo['U'+ctype]
-        typeobj = val[-1]
-        utypeobj = uval[-1]
-        if intname not in allTypes.keys():
-            uintname = 'uint%d' % bits
-            allTypes[intname] = typeobj
-            allTypes[uintname] = utypeobj
-            sctypeDict[intname] = typeobj
-            sctypeDict[uintname] = utypeobj
-            sctypeDict[Intname] = typeobj
-            sctypeDict[UIntname] = utypeobj
-            sctypeDict[charname] = typeobj
-            sctypeDict[ucharname] = utypeobj
-            sctypeNA[Intname] = typeobj
-            sctypeNA[UIntname] = utypeobj
-            sctypeNA[charname] = typeobj
-            sctypeNA[ucharname] = utypeobj
-        sctypeNA[typeobj] = Intname
-        sctypeNA[utypeobj] = UIntname
-        sctypeNA[val[0]] = Intname
-        sctypeNA[uval[0]] = UIntname
+        i_info = typeinfo[ctype]
+        u_info = typeinfo['U'+ctype]
+        bits = i_info.bits  # same for both
+
+        for info, charname, intname, Intname in [
+            (i_info,'i%d' % (bits//8,),  'int%d' % bits,  'Int%d' % bits),
+            (u_info,'u%d' % (bits//8,), 'uint%d' % bits, 'UInt%d' % bits)
+        ]:
```

Comment:
Might put this at the end of the previous line and indent both lines another 4 spaces.

Reply:
Also reasonable - feel free to edit in the web editor, because I'm not going to get around to checking out this branch today.

Out of curiousity, does PEP8 have anything to say here?

Reply:
It doesn't mention the `for` case explicitly, and the `if` version leaves it pretty open. I'm just thinking `C` here. It would also be reasonable to just indent the `]:`.